clients/spark/core/src/main/scala/io/treeverse/gc/DataLister.scala

```diff
@@ -25,3 +30,40 @@ class NaiveDataLister extends DataLister {
     dataList.toDF("address", "last_modified")
   }
 }
+
+class FileDescriptor(val path: String, val lastModified: Long) extends Serializable
+
+class ParallelDataLister extends DataLister with Serializable {
+  def listPath(
+      configMapper: ConfigMapper,
+      path: String
+  ): List[FileDescriptor] = {
+    val p = new Path(path)
+    val fs = p.getFileSystem(configMapper.configuration)
+    fs.listStatus(p)
+      .map((status: FileStatus) =>
+        new FileDescriptor(status.getPath.getName, status.getModificationTime)
+      )
+      .toList
+  }
+
+  override def listData(configMapper: ConfigMapper, path: Path): DataFrame = {
+    import spark.implicits._
+    val fs = path.getFileSystem(spark.sparkContext.hadoopConfiguration)
+    val slices = listPath(configMapper, path.toString)
```

Comment:
Shouldn't we exclude some paths?
1. Default `_lakefs`
2. `dummy`
3. Every other flat object that existed for previous repos.

Reply:
Filtering out is done outside the scope if this class.